python/dataset.cpp

```diff
@@ -89,21 +95,41 @@ void init_dataset(py::module &m) {
            py::arg("data") = std::map<std::string, Variable>{},
            py::arg("coords") = std::map<Dim, Variable>{},
            py::arg("labels") = std::map<std::string, Variable>{})
+      .def(py::init([](const DatasetProxy &other) { return Dataset{other}; }))
       .def("__setitem__", [](Dataset &self, const std::string &name,
                              Variable data) { self.setData(name, data); })
       .def("__setitem__",
            [](Dataset &self, const std::string &name,
               const DataConstProxy &data) { self.setData(name, data); })
+      // TODO: nvaytet: I do not understand why this is not covered by the
+      // py::implicitly_convertible<VariableProxy, Variable>();
+      // statement in variable.cpp, but this is needed if a VariableProxy is
+      // used instead of a Variable.
+      // Maybe it's because it's in a different file?
       .def("__setitem__",
-           [](Dataset &self, const std::string &name, const DataProxy &data) {
```

Comment:
Why has this overload disappeared?

Reply:
i thought the "not implemented yet" had been fixed by Igor's PR #375, and that `DataProxy` and `DataConstProxy` were implicitely comvertible so this was no longer required, but maybe I am mistaken?